adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -474,7 +490,26 @@ private AuthenticationResult postMessage(String requestMessage, Map<String, Stri
                 // Protocol related errors will read the error stream and report
                 // the error and error description
                 Logger.v(TAG, "Token request does not have exception");
-                result = processTokenResponse(response);
+                try {
+                    result = processTokenResponse(response);
+                } catch (final ServerRespondingWithRetryableException e) {
+                    if (mRetryOnce) {
+                        //retry once if it is a server responding with retryable error
+                        //500, 503 and 504 are the ones we retry
+                        mRetryOnce = false;
+                        try {
+                            Thread.sleep(mDelayTimePeriod);
+                        } catch (InterruptedException exception) {
+                            Log.e(TAG, "InterruptedException exception", exception);
```

Comment:
>  Log.e(TAG, "InterruptedException exception", exception); [](start = 26, length = 58)

consider giving a more descriptive error message. It's hard to tell what's actually happening if you see this in the log file when debugging customer issues. We have log something i.e. "InterruptedException occurs when having the thread sleep for some time due to the retry logic. "